src/modules/candy-machine/operationHandlers/initCandyMachineOperationHandler.ts

```diff
@@ -0,0 +1,47 @@
+import { Keypair } from '@solana/web3.js';
+import { Metaplex } from '../../../Metaplex';
+import { OperationHandler } from '../../../shared';
+import { InitCandyMachineOperation, InitCandyMachineOutput } from '../operations';
+import { initCandyMachineBuilder } from '../transactionBuilders';
+
+export const initCandyMachineOperationHandler: OperationHandler<InitCandyMachineOperation> = {
+  async handle(
+    operation: InitCandyMachineOperation,
+    metaplex: Metaplex
+  ): Promise<InitCandyMachineOutput> {
+    const { payer = metaplex.identity() } = operation.input;
+    const {
+      candyMachine = Keypair.generate(),
+      wallet = payer.publicKey,
+      authority = payer.publicKey,
+      candyMachineModel,
+      confirmOptions,
+    } = operation.input;
```

Comment:
No I cannot since that `payer` is used in the next destruct (so they cannot be in the same one as `payer` would not be assigned yet afaik).
Even if that somehow magically works I'd prefer to be explicit to say:
1. destruct payer
2. use it as default in the next destruct

Reply:
Yes, what I was trying to say is that it does work magically. Destructuring works sequentially. But I don't mind if you want to keep it as-is.